src/sphinxcontrib/towncrier/__init__.py

```diff
@@ -78,6 +78,12 @@ def _autodetect_scm_version():
         raise LookupError("Failed to locate the project's SCM repo")
 
 
+def iter_towncrier_files(fragments_dir: Path) -> Iterable:
+    for path in Path(fragments_dir).iterdir():
+        if path.is_file() and path.suffix == "rst":
```

Comment:
`"rst"` part also depends on the conventions. Technically, it's not necessary, it's totally optional and I only saw one or two projects requiring rst. The main requirement is to comply with towncrier categories and links to issues...

Reply:
Ok, this is definitely something we should ask towncrier then.

Or read every file (overdoing it is not costly)

Reply:
I think it's alright to match the files by the pattern. Here's how I do it in a GitHub App: https://github.com/sanitizers/chronographer-github-app/blob/04c024e/chronographer/event_handlers.py#L271-L306.